### PR TITLE
Adds new moodlets, psychotropic drugs

### LINKED ISSUE #41170: Suffocation moodlet doesn't work
## Reproduction:
1 .Throw yourself in space
2. Suffocate to death
3. Everything's fine, no moodlet

Here's the related code, it looks like it should work and I haven't been able to figure out what the issue is
https://github.com/tgstation/tgstation/blob/fe714bc8d4e9b69955f8e25672fa1aad3a380333/code/modules/mob/living/carbon/life.dm#L152-L173

### PR DESCRIPTION
**Pull request by Mickyan**
:cl: Mickyan
add: Added various new moodlets
add: Most notably, the Laughter chemical and breathing small quantities of N2O can improve mood
add: Psicodine: a new chemical that can restore sanity levels to normal and temporarily suppress phobias
add: Happiness: a homemade drug that supposedly gives a feeling of total bliss and.. emptiness? That can't be healthy... watch out for side effects.
fix: The suffocation moodlet now triggers correctly
fix: The drunk moodlet now clears as soon as the drunkenness wears off
/:cl:

Fixes #41170

List of new moodlets: 
Jittery 
Vomiting, vomiting all over yourself (gross)
Painful medicine application (silver sulfazine/styptic powder/syntflesh) - very short duration
Chemical euphoria (breathing N2O)
Laughter/Superlaughter reactions
Happiness drug related reactions

Psicodine: keeps your sanity at least above neutral level and suppresses phobias as long as it's in your system. Also available from pill bottles at the medivend at a premium.

Happiness: loosely inspired by Joy from LISA, very addictive, does minor brain damage, withdrawal makes you slowly go insane and overdose makes your mood jump all over the place. Also suppresses phobias. Good if you need a very quick and powerful boost but not very practical for keeping your mood up for long periods.

### CODE REVIEW

**Comment by Borisvanmemes** [comment]
I was going to say rename happiness to joy but then i read that it's already loosely based on it, good stuff.

**Comment by Mickyan** [comment]
I wanted to use joy but it didn't feel right without having the same effects

**Comment by Repukan** [comment]
I've got the joy...joy...joy in me

**Comment by Vidiveni** [comment]
Happiness should do more stuff, like turning all blood invisible, causing butterfly or rainbow hallucinations, making simplemob corpses look like piñatas, making ordinary corpses look like happy standing clowns (or whatever), adding a profanity filter to all chat (converting "fuck" to "love" etc). 

The possibilities are endless.

**Comment by GuyonBroadway** [comment]
Is psicodine gonna be that drug the stuttering weeaboo autists who seem to have a fetish for role-playing damaged people keep asking for.

"H-hi I have a p-p-perscrption f-for anti-depressants, c-c-could y-you f-f-fill it-t p-p-please?"


Also we should make a drug that is actually like joy. With the enhanced strength and healing and the too much makes you into something monsterous.

**Comment by davethwave** [comment]
Joy+people=drug addict zombies

**Comment by Repukan** [comment]
Any joy consumption turned people into monsters, not just overdosing.

**Comment by GuyonBroadway** [comment]
Nah it was explained in game that an overdose did trigger the transformation else several joy users in LISA would be mutants before the game began.

It's addictive properties and the intentional prolific distribution of joy drives people to overdose.

**Comment by fludd12** [comment]
Alternatively, it's Joy from We Happy Few, with similar mental effects but none of the monster transformation

**Comment by davethwave** [comment]
Might be fun if that you overdose on joy you get some flavor text saying, you forget x thing. It could be related to where you work in, such as a chemist forgetting how to use the console. Or even to a point if a traitor overdoses, the guy could forget the syndi passcode for the pda/pen/headset.

**Comment by subject217** [comment]
An easy cop-out for a joy monster would be the changeling horror form sprite. I don't think it would be very balanced though. If you make it shitty then it becomes something people will shoot up others with.

**Comment by GuyonBroadway** [comment]
In LISA there was a vaccine that prevented joy mutation. Perhaps getting some mutadone could stop the transformation.

**Comment by Repukan** [comment]
The only thing that stopped the mutation was taking more joy

**Comment by Repukan** [comment]
If I recall correctly

**Comment by GuyonBroadway** [comment]
According to the wiki its after you cross a certain dose threshold, extreme emotional and physical trauma or death will trigger it.

But that'd be a pain to do codewise for ss13. Best to just make it an overdose process thing.

File: code/datums/mood_events/drug_events.dm [hunk 1]
```diff
@@ -37,3 +37,16 @@
...
 /datum/mood_event/withdrawal_critical/add_effects(drug_name)
 	description = "<span class='boldwarning'>[drug_name]! [drug_name]! [drug_name]!</span>\n"
+
+/datum/mood_event/happiness_drug
+	description = "<span class='nicegreen'>I can't feel anything and I never want this to end.</span>\n"
+	mood_change = 50
+
+/datum/mood_event/happiness_drug_good_od
+	description = "<span class='nicegreen'>YES! YES!! YES!!!</span>\n"
+	mood_change = 100
```
Comment (ATH1909) [inline code/datums/mood_events/drug_events.dm]: The bad overdose moodlet times out; why doesn't the good overdose moodlet?

File: code/datums/mood_events/generic_negative_events.dm [hunk 2]
```diff
@@ -114,10 +114,29 @@
...
 	timeout = 1200
 
+/datum/mood_event/jittery
+	description = "<span class='warning'>I'm nervous and on edge and I can't stand still!!</span>\n"
+	mood_change = -2
+
+/datum/mood_event/vomit
+	description = "<span class='warning'>I just threw up. Gross.</span>\n"
+	mood_change = -2
+	timeout = 1200
```
Comment (ATH1909) [inline code/datums/mood_events/generic_negative_events.dm]: Welp, flypeople are gonna be pretty much perma-depressed now, as they need to vomit to eat (maybe they can inject themselves with nutriment? I'm not sure).

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm [hunk 3]
```diff
@@ -361,3 +361,84 @@
...
+		switch(reaction)
+			if(1)
+				M.emote("laugh")
+				SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "happiness_drug", /datum/mood_event/happiness_drug_good_od)
+			if(2)
+				M.emote("sway")
+				M.Dizzy(25)
+			if(3)
+				M.emote("frown")
+				SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "happiness_drug", /datum/mood_event/happiness_drug_bad_od)
```
Comment (ATH1909) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm]: Maybe you could have these OD moodlets cancel each other whenever they're applied (i.e. when you get the bad one, the good one immediately goes away (and the bad one goes into effect), and when you get the good one, the bad one immediately goes away (and the good one goes into effect)).

File: code/datums/mood_events/drug_events.dm (hunk 1 above)
Comment (Mickyan) [reply code/datums/mood_events/drug_events.dm]: oversight, I'll fix it

**Comment by ATH1909** [comment]
All in all, nice PR, but I'm worried about the first two things that I pointed out. This'll make chemistry (and ghetto chemistry) a bit more fun.

File: code/datums/mood_events/generic_negative_events.dm (hunk 2 above)
Comment (Mickyan) [reply code/datums/mood_events/generic_negative_events.dm]: it's a pretty negligible malus but I guess I'll add an exception for flypeople

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm (hunk 3 above)
Comment (Mickyan) [reply code/modules/reagents/chemistry/reagents/drug_reagents.dm]: that's how it works

Comment (ATH1909) [reply code/modules/reagents/chemistry/reagents/drug_reagents.dm]: Oh, my bad, then.

Comment (ATH1909) [reply code/modules/reagents/chemistry/reagents/drug_reagents.dm]: Wait, what I meant was not that their modifiers to mood would negate each other, but rather that if you had both the good overdose and bad overdose moodlets at the same time, you'd have a mood (bar other modifiers) of either +100 or -100 (NOT 0), depending on which one was applied most recently.

Comment (ATH1909) [reply code/modules/reagents/chemistry/reagents/drug_reagents.dm]: The bad moodlet takes 30 seconds to go away, but as far as I can tell, the good and the bad moodlets each have a 10% chance of going into effect each second. Thus, after the first few seconds, your mood should be mostly stable, as you'll almost always have both the bad and the good overdose moodlets.

Comment (Mickyan) [reply code/modules/reagents/chemistry/reagents/drug_reagents.dm]: The moodlets override each other, you can't have both at the same time

Comment (ATH1909) [reply code/modules/reagents/chemistry/reagents/drug_reagents.dm]: Oh, okay.

**Comment by GuyonBroadway** [comment]
Just as an aside, have you tried out what happens when you take this drug with moodlets turned off?

Assuming it does nothing might I suggest a snarky little quip along the lines of "you feel emotions but really aren't that fussed"

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 4]
```diff
@@ -1255,3 +1258,38 @@
...
+	..()
+
+/datum/reagent/medicine/psicodine/on_mob_life(mob/living/carbon/M)
+	M.jitteriness = max(0, M.jitteriness-6)
+	M.dizziness = max(0, M.dizziness-6)
+	M.confused = max(0, M.confused-6)
+	M.disgust = max(0, M.disgust-6)
+	GET_COMPONENT_FROM(mood, /datum/component/mood, M)
+	if (mood)
+		if(mood.sanity <= SANITY_NEUTRAL)
```
Comment (Qustinnus) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: reee use the wrappers for this ( setSanity() )

**Comment by Qustinnus** [comment]
@GuyonBroadway the beauty of SendSignal is that it doesn't do anything if the component doesnt exist, making it safe to send signals. Same goes with the manual sanity changing.

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm (hunk 4 above)
Comment (Mickyan) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: oh nice, that's perfect

**Comment by lolman360** [comment]
OD people on joy as a tator, turn them into murdery now-antag monsters that actually die when you kill them. It's like Romerol but better!

**Pull request merged**